cmd.cpp

```diff
@@ -218,4 +224,65 @@ void Dictionary::touch(string fileName, string currentFolder) {
 			cout << "Invalid filename" << endl;
 		}
 	}
-}
\ No newline at end of file
+	
+}
+
+void Dictionary::writeToFile(string fsname) {
+	ofstream systemStructure;
+	systemStructure.open(fsname + ".txt");
+	for (unsigned int i = 0; i < this->system.size(); i++) {
+		systemStructure << this->system[i].folder << "/" << this->system[i].subfolder << "\n";
+	}
+	systemStructure.close();
+}
+
+void Dictionary::loadFromFile(string fsname) {
+	string line;
+	Pair p;
+	ifstream systemStructure(fsname+".txt");
+	if (systemStructure.is_open())
+	{
+		while (getline(systemStructure, line))
+		{
+			p = this->splitStringFirstSlash(line);
+			this->system.push_back(p);
+		}
+		systemStructure.close();
+	}
+}
+
+void Dictionary::splitString(string &str, vector<string>& out, string delim) {
+	size_t start;
+	size_t end = 0;
+	while ((start = str.find_first_not_of(delim, end)) != std::string::npos)
+	{
+		end = str.find(delim, start);
+		out.push_back(str.substr(start, end - start));
+	}
+}
+
+void Dictionary::addPairToVector(string folder, string subfolder) {
+	Pair p;
+	p.folder = folder;
+	p.subfolder = subfolder;
+	this->system.push_back(p);
+}
+
+Pair Dictionary::splitStringFirstSlash(string line) {
+	Pair p;
+	string path = line;
+	int cut = line.find_last_of("/");
```

Comment:
szoval FirstSlash, erre find_last_of ? :-) 
De amugyis erre egy kulon fuggvenyt letrehozni kicsit overkill. ITt kb 2 sor az, ami a darabolassal megy el, a tobbi az, hogy valahogy belepakold ezt. De ha mindenkeppen ki van szedve, akkor lehetne mar eleve ennek a Pair-nek egy ilyen string alapu konstruktora, meg egy "toString" fuggvenye. Es akkor ez a logika relative kozel lenne egymashoz lekodolva. De ez csak otlet.

Reply:
True, nem a legjobb az elnevezés (megint), azt javítottam, és marad egyelőre ebben a formában, mert átláthatóbb. Az ötlet igazából nagyon jó, és azt akkor majd az egész kódra kéne alkalmaznunk, mert máshol is fölösleges ismétlődéseket spórolhatnánk vele. Ezt megvalósítom akkor.

Reply:
Konstruktor done. Todo, hogy a structot külön cpp-be kivegyük.

Reply:
Olvasgatom, amiket irsz, ha kerdes van, reagalok. Ha nincs kerdes, akkor ha nem irok, akkor ack-ztam. Arra kerlek, hogyha review ready ujra, akkor explicit irj a PR-hoz egy kommentet, amiben taggelsz, hogy nezzem meg ujra.

Reply:
Jah, es amit kesobbi PR-re akarsz elhalasztani, arrol mondjuk erdemes csinalni issue-t, hogy ne felejtsd el. Ha itt akarod megcsinalni, akkor tudsz itt ilyen todo list szeruseget is csinalni.